chalice/awsclient.py

```diff
@@ -483,3 +483,29 @@ def _client(self, service_name):
             self._client_cache[service_name] = self._session.create_client(
                 service_name)
         return self._client_cache[service_name]
+
+    def add_permission_for_authorizer(self, rest_api_id, function_arn,
+                                      random_id):
```

Comment:
I think it would.  IIRC correctly it was done this way to make testing easier.  I believe there were issues with the botocore stubber's `ANY` implementation.  Those might be fixed now, I can check.